src/config/views.py

```diff
@@ -6,15 +6,36 @@
 from backend.permissions import AdminOrAnonymousReadOnly
 
 
+class AllConfigView(APIView):
+    """
+    get:
+    List the backend configuration
+    """
+
+    throttle_scope = "config"
+    permission_classes = (AdminOrAnonymousReadOnly,)
+
+    def get(self, request):
+        if request.user.is_staff:
+            return FormattedResponse(config.get_all())
+        return FormattedResponse(config.get_all_non_sensitive())
+
 class ConfigView(APIView):
+    """
+    get:
+    Get a configuration option
+
+    post:
+    Update a configuration option
+
+    patch:
+    Update a configuration option
+    """
+
     throttle_scope = "config"
     permission_classes = (AdminOrAnonymousReadOnly,)
 
-    def get(self, request, name=None):
-        if name is None:
-            if request.user.is_staff:
-                return FormattedResponse(config.get_all())
-            return FormattedResponse(config.get_all_non_sensitive())
+    def get(self, request, name):
```

Comment:
I don't know how I feel about this change, I'm not convinced of the benefit of separating these out, I feel like theres a django rest framework abstraction to handle this case but idk. A more major issue that is actually blocking the merge is that this is vulnerable to CVE-2020-15235, you can skid the patch from https://github.com/ractf/core/commit/f3dc89b9f6ab1544a289b3efc06699b13d63e0bd

Reply:
The problem I was facing is that generateschema complained when there were multiple url patterns with the same view (as they had the same operation ID).

If you think they should still be merged I could try and sort something out?